central/reports/manager/manager_impl.go

```diff
@@ -23,6 +24,9 @@ type managerImpl struct {
 }
 
 func (m *managerImpl) Remove(ctx context.Context, id string) error {
+	if features.VulnMgmtReportingEnhancements.Enabled() {
```

Comment:
I want to understand what led to add feature flag check to all functions? Is the manager instance usage not guarded properly?

Reply:
The manager is used by v1 report configuration service and v1 report service in multiple functions. I did not want to disable the v1 APIs altogether just yet since V1 reporting is still visible in UI and I did not want someone accidentally click on it and see a 404 page.